### ISSUE TITLE
units(number) alias for nOut(number)

### ISSUE THREAD

**Issue by stolsvik** [deeplearning4j/deeplearning4j#3895]
When using the fluent API with ListBuilder, you actually do not need to specify the nIn(...) (number of neurons in).

The nOut(..) method then seems a tad strange. It would be better if it was named units(..), neurons(..), size(..), number(..). Keras uses "units".

With this change, along with #3888 and #3889, a config could look like this:

                .list()
                .inputType(InputType.feedForward(numInputs))
                .layer(new DenseLayer.Builder().units(1000).build())
                .layer(new DenseLayer.Builder().units(500).dropout(0.5d).build())
                .layer(new DenseLayer.Builder().units(1000).build())
                .layer(new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                        .units(numOutputs)
                        .activation(Activation.SOFTMAX)
                        .build())
                .build();

.. which reads very compact and fluently. It also more to the point describes that *this layer* has 1000 neurons, which I think fits more with how most people visualize a network in their heads.

**Comment by AlexDBlack** [comment]
Hm... I could probably be persuaded to add a .units(x) overload... Anyone else have an opinion on this?

**Comment by maxpumperla** [comment]
@AlexDBlack I'd personally be in favour of a `units` overload.

**Comment by tomthetrainer** [comment]
@AlexDBlack I would also be in favor of a `units` overload

**Comment by AlexDBlack** [comment]
https://github.com/deeplearning4j/deeplearning4j/pull/3900
